### PR TITLE
[kube-state-metrics] Add network policy and bump to 2.8.1

### PR DESCRIPTION
**Pull request by QuentinBisson**
#### What this PR does / why we need it

This PR adds support for environment constrained with network policies

#### Which issue this PR fixes

*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*

- fixes #

#### Checklist

- [x] [DCO](https://github.com/prometheus-community/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [x] Chart Version bumped
- [x] Title of the PR starts with chart name (e.g. `[prometheus-couchdb-exporter]`)

### CODE REVIEW

File: charts/kube-state-metrics/values.yaml [hunk 1]
```diff
@@ -181,6 +181,17 @@ podSecurityPolicy:
...
+## Configure network policy for kube-state-metrics
+networkPolicy:
+  enabled: false
+  # egress:
+  # - {}
+  # ingress:
+  # - {}
+  # podSelector:
+  #   matchLabels:
+  #     app: kube-state-metrics
```
Comment (mrueg) [inline charts/kube-state-metrics/values.yaml:193]: This should likely use the standardized app. kubernetes.io/name label

Comment (QuentinBisson) [reply charts/kube-state-metrics/values.yaml]: Sure, I changed that :)

**Comment by QuentinBisson** [comment]
@mrueg can you give it a look pretty please :)

**Review by mrueg — APPROVED** [review]

**Comment by mrueg** [comment]
thanks!

**Pull request merged**